drivers/homeless_summary_report/app/models/homeless_summary_report/report.rb

```diff
@@ -0,0 +1,484 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HomelessSummaryReport
+  class Report < SimpleReports::ReportInstance
+    include Filter::ControlSections
+    include Filter::FilterScopes
+    include Reporting::Status
+    include Rails.application.routes.url_helpers
+    include ActionView::Helpers::NumberHelper
+    include ArelHelper
+
+    acts_as_paranoid
+
+    belongs_to :user
+    has_many :clients
+
+    after_initialize :filter
+
+    # NOTE: this differs from viewable_by which looks at the report definitions
+    scope :visible_to, ->(user) do
+      return all if user.can_view_all_reports?
+      return where(user_id: user.id) if user.can_view_assigned_reports?
+
+      none
+    end
+
+    scope :ordered, -> do
+      order(updated_at: :desc)
+    end
+
+    def run_and_save!
+      start
+      create_universe
+      complete
+    end
+
+    def start
+      update(started_at: Time.current)
+    end
+
+    def complete
+      update(completed_at: Time.current)
+    end
+
+    def filter=(filter_object)
+      self.options = filter_object.to_h
+      # force reset the filter cache
+      @filter = nil
+      filter
+    end
+
+    def filter
+      @filter ||= begin
+        f = ::Filters::FilterBase.new(user_id: user_id)
+        f.update(options.with_indifferent_access) if options.present?
+        f
+      end
+    end
+
+    def self.viewable_by(user)
+      GrdaWarehouse::WarehouseReports::ReportDefinition.where(url: url).
+        viewable_by(user).exists?
+    end
+
+    def self.url
+      'homeless_summary_report/warehouse_reports/report'
+    end
+
+    def url
+      homeless_summary_report_warehouse_reports_report_url(host: ENV.fetch('FQDN'), id: id, protocol: 'https')
+    end
+
+    def title
+      _('Homeless Summary Report')
+    end
+
+    def multiple_project_types?
+      true
+    end
+
+    private def build_control_sections
+      # ensure filter has been set
+      filter
+      [
+        build_general_control_section(include_comparison_period: false),
+        build_hoh_control_section,
+        build_coc_control_section(true),
+      ]
+    end
+
+    def report_path_array
+      [
+        :homeless_summary_report,
+        :warehouse_reports,
+        :reports,
+      ]
+    end
+
+    private def available_support
```

Comment:
This method looks like it was copied from the Performance Metrics report.  Is it in use here?